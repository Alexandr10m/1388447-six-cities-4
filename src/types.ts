export interface LocalOffer {
  isPremium: boolean;
  pictures: string[];
  price: number;
  isFavourite: boolean;
  grade: number;
  title: string;
  type: string;
  bedroom: number;
  maxAdults: number;
  facilities: string[];
  coords: number[];
  locationZoom: number;
  id: number;
  description: string;
  previewImage: string;
  host: Host;
}

export interface CityOffers {
  city: string;
  cityCoords: number[];
  cityZoom: number;
  localOffers: LocalOffer[] | [];
}

export interface Host {
  id: number;
  name: string;
  isPro: boolean;
  avatarUrl: string;
}

export interface author {
  avatarUrl: string;
  email: string;
  id: number;
  isPro: boolean;
  name: string;
}

export interface Review {
  text: string;
  date: InstanceType<typeof Date>;
  id: number;
  grade: number;
  user: author;
}
