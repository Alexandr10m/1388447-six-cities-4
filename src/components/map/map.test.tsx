import * as React from "react";
import * as renderer from "react-test-renderer";
import Map from "./map";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import NameSpace from "../../reducer/name-space";
import {LocalOffer, CityOffers} from "../../types";


const mockStore = configureStore([]);

const offerWithPremium: LocalOffer = {
  bedroom: 2,
  coords: [48.865610000000004, 2.350499],
  description: `in city center.`,
  facilities: [`Air conditioning`, `Breakfast`],
  grade: 3.6,
  host: {
    avatarUrl: `img/avatar-angelina.jpg`,
    id: 25,
    isPro: true,
    name: `Angelina`,
  },
  id: 1,
  isFavourite: false,
  isPremium: true,
  locationZoom: 16,
  maxAdults: 8,
  pictures: [`https://`, `https`],
  previewImage: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`,
  price: 397,
  title: `Penthouse`,
  type: `hotel`,
};
const offerWithFavourite: LocalOffer = {
  bedroom: 2,
  coords: [48.865610000000004, 2.350499],
  description: `in city center.`,
  facilities: [`Air conditioning`, `Breakfast`],
  grade: 3.6,
  host: {
    avatarUrl: `img/avatar-angelina.jpg`,
    id: 25,
    isPro: true,
    name: `Angelina`,
  },
  id: 1,
  isFavourite: true,
  isPremium: false,
  locationZoom: 16,
  maxAdults: 8,
  pictures: [`https://`, `https`],
  previewImage: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`,
  price: 397,
  title: `Penthouse`,
  type: `hotel`,
};

const offers: CityOffers[] = [
  {
    city: `Paris`,
    cityCoords: [48.85661, 2.351499],
    cityZoom: 13,
    localOffers: [offerWithFavourite],
  },
  {
    city: `Amsterdam`,
    cityCoords: [52.38333, 4.9],
    cityZoom: 13,
    localOffers: [offerWithPremium],
  }
];

const store = mockStore({
  [NameSpace.STATE]: {
    sortType: `Popular`,
    indicatedCard: offerWithPremium,
    city: `Paris`,
  },
  [NameSpace.DATA]: {
    offers
  },
});

const props = {
  localOffers: [
    offerWithPremium,
    offerWithFavourite,
  ],
  cityCoords: [48.85661, 2.351499],
  cityZoom: 13,
  locationZoom: 16,
  activeCard: offerWithPremium,
};

describe(`Snapshot of Map`, () => {
  it(`Map should render correctly`, () => {
    jest.mock(`leaflet`);

    const tree = renderer
      .create(
          <Provider store={store}>
            <Map
              {...props}
            />
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
