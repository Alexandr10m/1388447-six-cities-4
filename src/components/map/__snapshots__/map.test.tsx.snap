// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Map Map should render correctly 1`] = `
<div
  id="map"
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
/>
`;
